plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/codeInsight/unwrap/KotlinFunctionParameterUnwrapper.kt

```diff
@@ -3,25 +3,176 @@
 package org.jetbrains.kotlin.idea.codeInsight.unwrap
 
 import com.intellij.psi.PsiElement
-import org.jetbrains.kotlin.psi.KtCallExpression
-import org.jetbrains.kotlin.psi.KtValueArgument
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.util.elementType
+import org.jetbrains.kotlin.idea.KotlinBundle.message
+import org.jetbrains.kotlin.idea.refactoring.getExpressionShortText
+import org.jetbrains.kotlin.idea.util.ifTrue
+import org.jetbrains.kotlin.idea.util.isComma
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.getNextSiblingIgnoringWhitespaceAndComments
+import org.jetbrains.kotlin.psi.psiUtil.getPrevSiblingIgnoringWhitespaceAndComments
 import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
 
-class KotlinFunctionParameterUnwrapper(key: String) : KotlinUnwrapRemoveBase(key) {
+class KotlinFunctionParameterUnwrapper(val key: String) : KotlinUnwrapRemoveBase(key) {
 
     override fun isApplicableTo(element: PsiElement): Boolean {
-        if (element !is KtCallExpression) return false
-        if (element.valueArguments.size != 1) return false
-        val argument = element.parent as? KtValueArgument ?: return false
-        if (argument.getStrictParentOfType<KtCallExpression>() == null) return false
+        val argumentToUnwrap = argumentToUnwrap(element) ?: return false
+        deletionTarget(argumentToUnwrap) ?: return false
+
+        return true
+    }
+
+    private fun argumentToUnwrap(element: PsiElement?): KtValueArgument? {
+        if (element == null) return null
+
+        nearbySingleArg(element)?.let {
+            return it
+        }
+        when (element) {
+            // KtLambdaArgument is also a KtValueArgument, but it's already handled by KotlinLambdaUnwrapper
```

Comment:
It's not true that `KtLambdaArgument` is already handled by `KotlinLambdaUnwrapper`.

`KotlinLambdaUnwrapper` should unwrap it this way:
```
run { println("Hi") } -> println("Hi")
```

`KotlinFunctionParameterUnwrapper` should unwrap it this way:
```
run { println("Hi") } -> { println("Hi") }
```

Reply:
You certainly took a good look at my change, thanks :)